certbot-apache/certbot_apache/_internal/configurator.py

```diff
@@ -706,6 +706,15 @@ def choose_vhost(self, target_name, create_if_no_ssl=True):
         :raises .errors.PluginError: If no vhost is available or chosen
 
         """
+
+        if util.is_ipaddress(target_name):
+            # ip address currently doesn't work for install
+            logger.error(
+                "Currently plugin doesn't support installing cert for ip address %s",
```

Comment:
f-strings are supported. I'm not sure how the maintainers feel about them, though. https://realpython.com/python-f-strings/

Reply:
That may because f-strings were introduced in 3.6, and that became the minimum supported version only just recently. f-strings are faster, and IMO read nicer. `f"hello {world}"` vs `"hello {}".format(world)`".

Reply:
https://github.com/certbot/certbot/blob/master/certbot/setup.py has some use of f-strings :)